pkgs/development/python-modules/pydiscourse/default.nix

```
@@ -0,0 +1,34 @@
+{
+  buildPythonPackage,
+  fetchFromGitHub ,
+  lib,
+  pythonAtLeast,
+  requests,
+}:
+
+buildPythonPackage rec {
+  pname = "pydiscourse";
+  version = "1.4.0";
+  format = "setuptools";
+
+  # the tox.ini file sets the tests up for Python 3.[7-10] only
+  patches = [ ./0001-Enable-the-tests-for-Python-3.11.patch ];
```

The patch is not needed. It looks like that the tests are not sensitive to the Python release. `unittestCheckHook` will help here.

```suggestion
  nativeCheckInputs = [
    unittestCheckHook
  ];
```